importlib_metadata/__init__.py

```diff
@@ -1089,11 +1091,14 @@ def _get_toplevel_name(name: PackagePath) -> str:
     >>> _get_toplevel_name(PackagePath('foo.dist-info'))
     'foo.dist-info'
     """
-    return _topmost(name) or (
-        # python/typeshed#10328
-        inspect.getmodulename(name)  # type: ignore
-        or str(name)
-    )
+    n = _topmost(name)
+    if n:
+        return n
+
+    global inspect
+    if inspect is None:
+        import inspect
```

Comment:
Since we don't have a test protecting this behavior, I'd like to see a comment pointing to the issue, so a future someone doesn't refactor this optimization away.